src/mscorlib/src/System/Delegate.cs

```diff
@@ -179,7 +179,7 @@ public override int GetHashCode()
             else
                 return unchecked((int)((long)this._methodPtrAux));
             */
-            return GetType().GetHashCode();
+            return ( _target != null ? _target.GetHashCode() : 0) * 33 + GetType.GetHashCode();
```

Comment:
`_target` should be included in the hash code only if  `_methodPtrAux` is null.

`_target` is overloaded field. It points other helper delegate objects when _methodPtrAux is non-null. The table at https://github.com/dotnet/coreclr/blob/5e03cb09e3754e4a3430ba424a1352b9c01ff098/src/vm/comdelegate.cpp#L3471 has the summary of the different options.